ext/re2/re2.cc

```diff
@@ -1589,7 +1589,13 @@ static VALUE re2_set_add(VALUE self, VALUE pattern) {
 
   int index = s->set->Add(regex, &err);
   if (index < 0) {
-    rb_raise(rb_eArgError, "str rejected by RE2::Set->Add(): %s", err.c_str());
+    char msg[100];
+    snprintf(msg, sizeof(msg), "str rejected by RE2::Set->Add(): %s",
+        err.c_str());
+
+    /* Manually destruct the error string before we throw an exception. */
+    err.~basic_string();
```

Comment:
Why is this needed? I would have thought once `err` goes out of scope that would be sufficient to free the `std::string`.

Reply:
Sadly not, it seems `rb_raise`'s use of `longjmp` prevents C++ destructors from working as expected.

That said, I think I have a nicer way to do this without explicitly calling the destructor by adding a block scope around the call to `Add()`.

Reply:
Updated in https://github.com/mudge/re2/pull/105/commits/152d4780cef3bd2995b22ce1011a86d9ef6c2c37 to no longer explicitly call the destructor but let a block scope do that for us.

Reply:
Great, I like this solution better.